Source/ThirdParty/ANGLE/src/libANGLE/Framebuffer.cpp

```diff
@@ -2896,4 +2900,39 @@ angle::Result Framebuffer::syncAttachmentState(const Context *context,
 
     return angle::Result::Continue;
 }
+
+void Framebuffer::markDrawAttachmentsNeedInit(size_t count, const GLenum *attachments)
+{
+    for (size_t attachmentIdx = 0; attachmentIdx < count; attachmentIdx++)
+    {
+        GLenum attachmentBindPoint = attachments[attachmentIdx];
+        switch (attachmentBindPoint)
+        {
+        case GL_DEPTH_ATTACHMENT:
+        case GL_DEPTH_EXT:
+            if (mState.mDepthAttachment.isAttached())
+            {
+                mState.mResourceNeedsInit.set(DIRTY_BIT_DEPTH_ATTACHMENT);
```

Comment:
Do you need to mark the resource itself as needing init?

E.g. if the resource is read or written through this Framebuffer, the framebuffer state mResourceNeedsInit should probably take care of the initialization.

However, if the resource is bound to some other fbo and then read through it, or  read through it some other means (shader samplling), then ANGLE cannot understand to clear it?

Does this need   `attachment.setInitState(InitState::MayNeedInit);` per the attachment?

Reply:
It also needs to notify the framebuffer observers about the dirty state.